src/OrganisationRegistry.Api/Security/ClaimsExtension.cs

```diff
@@ -15,11 +16,22 @@ public static void AddOrUpdateClaim(this ClaimsIdentity identity, string key, Cl
             identity.AddClaim(claim);
         }
 
-        public static string GetClaim(this ClaimsPrincipal user, string claimType) => user.Claims.SingleOrDefault(x => x.Type == claimType)?.Value;
-        public static string GetClaim(this ClaimsIdentity identity, string claimType) => identity.Claims.SingleOrDefault(x => x.Type == claimType)?.Value;
+        public static string? GetOptionalClaim(this ClaimsPrincipal user, string claimType)
+            => user.Claims.SingleOrDefault(x => x.Type == claimType)?.Value;
 
-        public static IEnumerable<string> GetClaims(this ClaimsPrincipal user, string claimType) => user.Claims.Where(x => x.Type == claimType).Select(x => x.Value);
-        public static IEnumerable<string> GetClaims(this ClaimsIdentity identity, string claimType) => identity.Claims.Where(x => x.Type == claimType).Select(x => x.Value);
+        public static string? GetOptionalClaim(this ClaimsIdentity identity, string claimType)
+            => identity.Claims.SingleOrDefault(x => x.Type == claimType)?.Value;
 
+        public static string GetRequiredClaim(this ClaimsPrincipal user, string claimType)
+            => user.Claims.Single(x => x.Type == claimType).Value;
+
+        public static string GetRequiredClaim(this ClaimsIdentity identity, string claimType)
+            => identity.Claims.Single(x => x.Type == claimType).Value;
+
+        public static IEnumerable<string> GetClaims(this ClaimsPrincipal user, string claimType)
```

Comment:
ImmutableList<string>

Reply:
I want to leave this open to the caller to make this into an immutable list.